### ISSUE TITLE
Get page size fix into Debian 9

### ISSUE THREAD

**Issue by vielmetti** [vielmetti/go-pagesize-test#5]
Page size bug is in the Go 1.7.4 that ships with Debian 9 as delivered on Packet.

**Comment by vielmetti** [comment]
go version go1.7.4 linux/arm64

**Comment by vielmetti** [comment]
```
go-pagesize-test# ./go-pagesize-test 
OS page size = 4096 ; go reports 65536
```

**Comment by vielmetti** [comment]
upgraded to `testing` version

`go version go1.7.6 linux/arm64`

problem persists.

**Comment by zumbi** [comment]
I am trying to reproduce this issue, and I am able to reproduce it, however if I apply on top of go1.7.4 the suggested patch (1b9499b06989d2831e5b156161d6c07642926ee1) to test the fix, I get the following issue:

/home/zumbi/golang-1.7-1.7.4/src/runtime/runtime.go:56: undefined: physPageSize

Regards

**Comment by zumbi** [comment]
On SID environment with golang1.9, the test passes.
```
(sid_arm64-dchroot)zumbi@amdahl:~/go-pagesize-test$ make
go build go-pagesize-test.go
bats go-pagesize-test.bats
 ✓ go reports correct pagesize

1 test, 0 failures
```

**Comment by dato** [comment]
The fix above cannot be backported to 1.7, because support for page size detection at runtime was only introduced in Go 1.8 (golang/go@276a52de55fb48c4e56a778f1f7cac9292d8fad7). The fix above builds on that, which is why you get the error about `physPageSize` not being defined.

Whatever this bug is blocking in Debian 9, it will be easier to take the same approach of Ubuntu in their [16.04 fix](https://launchpad.net/ubuntu/+source/golang-1.6/1.6.2-0ubuntu5~16.04.4), which is to change the hard-coded value to whatever is appropriate for the kernels shipped by the distribution.  (This, of course, depending on the configuration of shipped kernels; for Ubuntu [it was okay](https://bugs.launchpad.net/ubuntu/+source/golang-defaults/+bug/1661222/comments/10).)

**Comment by vielmetti** [comment]
Agreed with @dato that if Debian 9 has a characteristic page size, that the resolution for this issue should be as simple as hard-coding the default to the characteristic page size used by the distribution.

**Comment by vielmetti** [comment]
I've updated the README.md file to point at the way that Ubuntu fixed this in Ubuntu 16.04 LTS, and I've added the Dockerfile test to show the impact on deeply nested filesystems.
